docs/Using-Fleet/REST-API.md

```diff
@@ -5480,6 +5490,114 @@ load balancer timeout.
+| --------------- | ------  | ----- | ----------------------------------------------------------------------------------------------------------------------------- |
+| id              | integer | path  | **Required** The ID of the parent team.
+| order_key       | string  | query | What to order results by. Can be any column in the queries table.                                                             |
+| order_direction | string  | query | **Requires `order_key`**. The direction of the order given the order key. Options include `asc` and `desc`. Default is `asc`. |
+
+#### Example
+
+`GET /api/v1/fleet/team/123/queries`
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+"queries": [
+  {
+    "created_at": "2021-01-04T21:19:57Z",
+    "updated_at": "2021-01-04T21:19:57Z",
+    "id": 1,
+    "name": "query1",
+    "team_id": 123,
+    "description": "query",
+    "query": "SELECT * FROM osquery_info",
+    "saved": true,
+    "observer_can_run": true,
+    "author_id": 1,
+    "author_name": "noah",
+    "author_email": "noah@example.com",
+    "packs": [
+      {
+        "created_at": "2021-01-05T21:13:04Z",
+        "updated_at": "2021-01-07T19:12:54Z",
+        "id": 1,
+        "name": "Pack",
+        "description": "Pack",
+        "platform": "",
+        "disabled": true
+      }
+    ],
+    "stats": {
+      "system_time_p50": 1.32,
+      "system_time_p95": 4.02,
+      "user_time_p50": 3.55,
+      "user_time_p95": 3.00,
+      "total_executions": 3920
+    }
+  },
+  {
+    "created_at": "2021-01-19T17:08:24Z",
+    "updated_at": "2021-01-19T17:08:24Z",
+    "id": 3,
+    "name": "osquery_schedule",
+    "description": "Report performance stats for each file in the query schedule.",
+    "team_id": 123,
+    "query": "select name, interval, executions, output_size, wall_time, (user_time/executions) as avg_user_time, (system_time/executions) as avg_system_time, average_memory, last_executed from osquery_schedule;",
+    "saved": true,
+    "observer_can_run": true,
+    "author_id": 1,
+    "author_name": "noah",
+    "author_email": "noah@example.com",
+    "packs": [
+      {
+        "created_at": "2021-01-19T17:08:31Z",
+        "updated_at": "2021-01-19T17:08:31Z",
+        "id": 14,
+        "name": "test_pack",
+        "description": "",
+        "platform": "",
+        "disabled": false
+      }
+    ]
+  }
+]}
+```
+
+### Get team query
+Use the [get query](#get-query) endpoint, which supports both global queries and team queries.
+
+### Create team query
+Use the [create query](#create-query) endpoint, and submit a `team_id` attribute in the body.
+
+### Modify team query
+Use the [modify query](#modify-query) endpoint, which supports both global queries and team queries.
+
+### Delete team query
+Use the [delete query](#delete-query) endpoint, which supports both global queries and team queries.
+
```

Comment:
The convention we've established with the `/schedules/`, `/packs/`, and `/policies/` endpoints is that we have dedicated endpoints for team operations. I'd like to maintain that convention versus the query endpoints operating differently than the rest of the API. That would mean we would have new endpoints like: 

- `GET /teams/{team_id}/queries/{query_id}`
- `POST /teams/{team_id}/queries/{query_id}`
- `PATCH /teams/{team_id}/queries/{query_id}`
- `DELETE /teams/{team_id}/queries/{query_id}`

There's a good argument to be made for using fewer API endpoints and passing a `team_id` param, but I think that is a conversation for v5 and breaking API changes.

Reply:
ok i'll change it to match the convention.

But this approach wouldn't cause a breaking change would it? I was thinking you submit a team_id and it would create a query on a team, but if you don't pass in a team_id, then it would create a global query, as before.

Reply:
It would not be a breaking change. It's a good suggestion worth considering for v5. If we _really_ wanted to do it in V4, we could keep all the old endpoints and add the same `team_id` behavior to the existing global endpoints for consistency, but I don't think it'd be worth the time investment right now.